File: exercism/javascript/clock/clock.js
export function at(hour, minute) {
  const hoursAsMinutesFromMidnight = (24 + (hour % 24) || 0) * 60;
  const minutesInTheHour = minute || 0;
  const time = hoursAsMinutesFromMidnight + minutesInTheHour;

  return {
    plus() {
      throw new Error("Remove this statement and implement this function");
    },

    minus() {
      throw new Error("Remove this statement and implement this function");
    },

    equals() {
      throw new Error("Remove this statement and implement this function");
    },

    toString() {
      const hours = `${Math.floor((time / 60) % 24)}`.padStart(2, "0");
      const minutes = `${time % 60}`.padStart(2, "0");

      return `${hours}:${minutes}`;
    }
  };
}

File: exercism/python/clock/clock.py
class Clock():
    def __init__(self, hour, minute):
        self._minutes = minute + hour * 60

    def hour(self):
        return (self._minutes // 60) % 24

    def minute(self):
        return self._minutes % 60

    def add(self, minutes):
        self._minutes += minutes
        return self

    def __str__(self):
        return '{:{fill}2}:{:{fill}2}'.format(
            self.hour(), self.minute(), fill='0')

    def __eq__(self, other):
        return self.hour() == other.hour() and self.minute() == other.minute()
